images/vmi-router/controllers/vmirouter_reconciler.go

```diff
@@ -73,29 +74,24 @@ func (r *VMRouterReconciler) Reconcile(ctx context.Context, req reconcile.Reques
 	r.log.V(4).Info(fmt.Sprintf("Got reconcile request for %s", req.String()))
 
 	// Start with retrieving affected VMI.
-	var vm virtv1alpha2.VirtualMachine
-	var isAbsent bool
-	err := r.client.Get(ctx, req.NamespacedName, &vm)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			isAbsent = true
-		} else {
-			r.log.Error(err, fmt.Sprintf("fail to retrieve vm/%s", req.String()))
-			return reconcile.Result{}, err
-		}
+	vm := &virtv1alpha2.VirtualMachine{}
```

Comment:
Usually, if it's a pointer, we want to check it for nil. However, in the case of such a definition, vm == nil will always be false because it's a pointer to an empty resource, not nil.